plugins/meta/sbr/README.md

```diff
@@ -0,0 +1,126 @@
+way of telling whether a destination IP on the internet is (say) an address
+used for downloading updated software packages or a remote SIP endpoint.
+
+Hence Source Based Routing is used.
+
+- The application explicitly listens on the correct interface for incoming
+  traffic.
+
+- When the application wishes to send to an address via the SIP network, it
+  explicitly binds to the IP of the device on that network.
+
+- Routes for the SIP interface are configured in a separate routing table, and
+  a rule is configured to use that table based on the source IP address.
+
+Note that in most cases there is a management device (the first one) and that
+the SBR plugin is called once for each device after the first, leaving the
+default routing table applied to the management device. However, this not
+mandatory, and source based routing may be configured on any or all of the
+devices as appropriate.
+
+## Usage
+
+This plugin runs as a chained plugin, and requires the following information
+passed in from the previous plugin (which has just set up the network device):
+
+- What is the network interface in question?
+
+- What is the IP address (or addresses) of that network interface?
+
+- What is the default gateway for that interface (if any)?
+
+It then reads all routes to the network interface in use.
+
+Here is an example of what the plugin would do. (The `ip` based commands are
+implemented in go, but easier to describe via the command line.) Suppose that
+it reads that:
+
+- The interface is `net1`.
+
+- The IP address on that interface is `192.168.1.209`.
+
+- The default gateway on that interface is `192.168.1.1`.
+
+- There is one route configured on that network interface, which is
+  `192.168.1.0/24`.
+
+Then the actions it takes are the following.
+
+- It creates a new routing table, and sets a rule to use it for the IP address in question.
+
+        ip rule add from 192.168.1.209/32 table 100
+
+- It adds a route to the default gateway for the relevant table.
+
+        ip route add default via 192.168.1.1 dev net1 table 100
+
+- It moves every existing route on the device to the new table.
+
+        ip route del 192.168.1.0/24 dev net1 src 192.168.1.209
+        ip route add 192.168.1.0/24 dev net1 src 192.168.1.209 table 100
+
+On deletion it:
+
+- deletes the rule (`ip rule del from 192.168.1.209/32 table 100`), which it
+  finds by deleting rules relating to IPs on the device which is about to be
+  deleted.
+
+- does nothing with routes (since the kernel automatically removes routes when
+  the device with which they are associated is deleted).
+
+## Future enhancements and known limitations
+
+The following are possible future enhancements.
+
+- The table number is currently selected by starting at 100, then incrementing
+  the value until an unused table number is found. It might be nice to have an
+  option to pass the table number as an input.
+
+- The logging is entirely on or entirely off - there is no log severity.
```

Comment:
I can't see a way to turn the logging off?  Maybe I'm missing something.  Would be nice to be able to change.

Reply:
Or if there isn't a way to turn it off, I think we're fine with fixing the README file and merging, right Matt?

Reply:
There is no way to turn it off. There used to be logging to file, and when I removed that capability, I missed changing the README. Will fix this now.